src/goto-instrument/loop_utils.cpp

```diff
@@ -38,19 +38,28 @@ void build_havoc_code(
   const goto_programt::targett loop_head,
   const modifiest &modifies,
   goto_programt &dest)
+{
+  build_havoc_code_at_source_location(
+    loop_head->source_location, modifies, dest);
+}
+
+void build_havoc_code_at_source_location(
+  const source_locationt source_location,
+  const modifiest &modifies,
+  goto_programt &dest)
 {
   for(modifiest::const_iterator
       m_it=modifies.begin();
       m_it!=modifies.end();
       m_it++)
   {
     exprt lhs=*m_it;
-    side_effect_expr_nondett rhs(lhs.type(), loop_head->source_location);
+    side_effect_expr_nondett rhs(lhs.type(), source_location);
```

Comment:
Hopefully this was its own commit? I'd like to know more about what the changes in this file are for, is it a bugfix? I don't see any tests

Reply:
They were accidentaly included here. Thanks for the catch